### ISSUE TITLE
[9.4 Staging 1987] Skill UI Bug

### ISSUE THREAD

**Issue by TwixterZ** [StrangeLoopGames/EcoIssues#21924]
Skill UI is not working as intended.

Repro :
1 - On fresh world, open skill UI window.

You will be greeted with below window :
![image.png](https://images.zenhubusercontent.com/6017bedcbf0e6bfb14e12003/b1f7c2ad-8bcd-4fe1-a6b9-8a93504c08a8)

2 - Click on logging skill will not do anything
3 - Click on other skills, UI will be showing as normal again.
4 - Now, try to click on logging again
5 - Logging icon will be highlighted (yellow frame) but composite is still selected.
![image.png](https://images.zenhubusercontent.com/6017bedcbf0e6bfb14e12003/3d52ec1d-7b21-49ca-8068-9c0d0c48fa13)

_This issue is constant with all skills that is already unlocked/discovered_
_you may use /skill give *any skill* and attempt to click said skill in UI and will face same issue._

**Comment by SlayksWood** [comment]
[Player.log](https://github.com/StrangeLoopGames/EcoIssues/files/6513877/Player.log)
```
NullReferenceException: Object reference not set to an instance of an object.
  at LearningOptionUI.get_StarReqCount () [0x00000] in <00000000000000000000000000000000>:0 
  at LearningOptionUI.UpdateReqs () [0x00000] in <00000000000000000000000000000000>:0 
  at System.Threading.ThreadStart.Invoke () [0x00000] in <00000000000000000000000000000000>:0 
  at Eco.Shared.View.SubscriptionsExtensions.SubscribeAndCall (Eco.Shared.View.ISubscriptions ss, Eco.Shared.View.IObservable s, System.String propname, System.Action changedCallback) [0x00000] in <00000000000000000000000000000000>:0 
  at LearningOptionUI.Setup (SkillView skill, System.Boolean selfTeachOption) [0x00000] in <00000000000000000000000000000000>:0 
  at SpecialtyWindowClaiming.Setup () [0x00000] in <00000000000000000000000000000000>:0 
  at Assets.UI.Scripts.Skills.SpecialtyWindowUI.Init (SkillTreeView skillTree) [0x00000] in <00000000000000000000000000000000>:0 
  at SkillWindowUI.Open (Eco.Shared.Serialization.BSONObject bsonObj) [0x00000] in <00000000000000000000000000000000>:0 
  at UI.UIManager.Open (System.String guiName, Eco.Shared.Serialization.BSONObject bson, UI.UILayer layer, System.Boolean singleInstance) [0x00000] in <00000000000000000000000000000000>:0 
  at UI.UIManager.Toggle (System.String guiName, Eco.Shared.Serialization.BSONObject bson, UI.UILayer layer) [0x00000] in <00000000000000000000000000000000>:0 
  at UI.UIManager.Toggle (System.String guiName, UI.UILayer layer) [0x00000] in <00000000000000000000000000000000>:0 
  at SkillsButton.<Initialize>b__11_0 () [0x00000] in <00000000000000000000000000000000>:0 
  at RootMotion.Dynamics.PuppetMaster+UpdateDelegate.Invoke () [0x00000] in <00000000000000000000000000000000>:0 
  at UnityEngine.Events.InvokableCall.Invoke (System.Object[] args) [0x00000] in <00000000000000000000000000000000>:0 
  at UnityEngine.Events.UnityEvent.Invoke () [0x00000] in <00000000000000000000000000000000>:0 
  at SkillsButton.ExternalUpdate () [0x00000] in <00000000000000000000000000000000>:0 
  at ActionBarUI.Update () [0x00000] in <00000000000000000000000000000000>:0 
UnityEngine.Logger:LogException(Exception, Object)
UnityEngine.Debug:CallOverridenDebugHandler(Exception, Object)
```

**Comment by SlayksWood** [comment]
test this one after fix #20390

**Comment by SlayksWood** [comment]
blocks a lot of game features to test, up to critical
